oidc_cli/storage/storage.go

```diff
@@ -20,3 +23,31 @@ type Storage interface {
 
 	MarshalOpts() []client.MarshalOpts
 }
+
+func GetStorage(clientID string, issuerURL string) (Storage, error) {
+	isWSL, err := osutil.IsWSL()
+	if err != nil {
+		return nil, err
+	}
+
+	// If WSL we use a file storage which does not cache refreshTokens
+	//    we do this because WSL doesn't have a graphical interface
+	//    and therefore limits how we can interact with a keyring (such as gnome-keyring).
+	// To limit the risks of having a long-lived refresh token around,
+	//    we disable this part of the flow for WSL. This could change in the future
+	//    when we find a better way to work with a WSL secure storage.
+	if isWSL {
+		return getFileStorage(clientID, issuerURL)
+	}
```

Comment:
There's a few things being conflated here:

- We're checking for WSL specifically
- We don't really care about WSL per-say, just that it:
  - doesn't have a GUI
  - doesn't have a secure way to save tokens

Correct me if I'm wrong, I think all we _really_ care about here is if the client has a way to safely store credentials. I'd be curious to see if we can figure out a better way to check for secure storage mechanisms, or if client OS checking is the best we can do.

Reply:
@edulop91 Do you have ideas here? I just wanted to preserve the way we handled this when copying it over, to not break anything.

Reply:
yeah, those are good suggestions. we should probably save those improvements for a separate PR so we're keeping PR/changes focused though